tutorials/how_to/span_representations.md

```diff
@@ -0,0 +1,180 @@
+from allennlp.data.tokenizers.word_tokenizer import SpacyWordSplitter
+from allennlp.data.tokenizers.token import Token
+
+tokenizer = SpacyWordSplitter(pos_tags=True)
+sentence = tokenizer.split_words("This is a sentence.")
+
+def no_prefixed_punctuation(tokens: List[Token]) -> bool:
+    # Only include spans which don't start or end with punctuation.
+    return tokens[0].pos_ != "PUNCT" and tokens[-1].pos_ != "PUNCT"
+
+spans = span_utils.enumerate_spans(sentence,
+                                    max_span_width=3,
+                                    min_span_width=2,
+                                    filter_function=no_prefixed_punctuation)
+
+# 'spans' won't include (2, 4) or (3, 4) as these have
+# punctuation as their last element. Note that these spans
+# have inclusive start and end indices!
+assert spans == [(0, 1), (0, 2), (1, 2), (1, 3), (2, 3)]
+```
+
+There are other helpful functions in `allennlp.data.dataset_readers.dataset_utils.span_utils`,
+such as a function to convert between BIO labelings and span based representations.
+
+### Use a SpanPruner
+
+It's not always possible to prune spans before they enter your model. AllenNLP contains
+a `SpanPruner`, which allows you to prune spans based on a parameterised function which
+is trained end-to-end with the rest of your model.
+
+```
+import torch
+from torch.autograd import Variable
+from allennlp.modules import SpanPruner
+
+# Create a linear layer which will score our spans.
+linear_scorer = torch.nn.Linear(5, 1)
+pruner = SpanPruner(scorer=linear_scorer)
+
+# Here we'll create some spans from a random tensor of shape
+# (batch_size, num_spans, embedding_size). Typically this would
+# be the output of a SpanExtractor applied to some encoded representation
+# of a sentence, such as the output of an LSTM, or word embeddings.
+spans = Variable(torch.randn([3, 4, 5]))
+mask = Variable(torch.ones([3, 4]))
+
+# Ok, there's quite a bit to unpack here.
+# See below for a full explaination.
+pruned_embeddings, pruned_mask, pruned_indices, pruned_scores = pruner(spans, mask, num_spans_to_keep=3)
+```
+
+1. First, we've got our `pruned_embeddings`. 
+These are of shape `(batch_size, num_spans_to_keep, embedding_size)`
+The spans we kept corespond to the topk with respect to the parameterised
+span scorer. The other spans just get discarded, and your eventual loss
+function for your model won't be a function of the discarded spans!
+
+2. Secondly, we've got the `pruned_mask`, which has shape `(batch_size, num_spans_to_keep)`.
+In 99% of cases, this will be all ones. However, if you have masked spans in a 
+batch element, and you request that the `SpanPruner` keeps more than the number
+of non-masked spans, there will be some masked elements in the returned spans.
+
+3. Thirdly, we have the `pruned_indices` which has shape `(batch_size, num_spans_to_keep)`
+The indices of the top-k scoring spans into the original ``spans`` tensor. 
```

Comment:
The hyphen in "top-k" makes me think the bracketing is "(top-k) (scoring spans)", instead of "(top k scoring) spans".  I'd also probably use "in" instead of "into" here.

Reply:
I feel like you need some sort of conjunction between these two sentences